lib/internal/http2/core.js

```diff
@@ -1622,6 +1627,13 @@ class ClientHttp2Session extends Http2Session {
 
     this[kUpdateTimer]();
 
+    for (const i in headers) {
+      if (i[0] === ':') {
+        assertValidPseudoHeader(i);
+      } else if (i && !checkIsHttpToken(i))
+        this.destroy(new ERR_INVALID_HTTP_TOKEN('Header name', i));
+    }
```

Comment:
@rexagod I kind of understand why we’re doing this for the compatibility API – to match HTTP/1 behaviour – but I don’t quite understand why we’re doing this here as well. Aren’t HTTP/2 header names basically arbitrary? Can you clarify that for me?

Reply:
No, they still are supposed to follow the http spec in terms of construction and what characters are allowed or not allowed. Http2 Just makes it safer to encode those just in case there are invalid characters.